tests/state/test_state_api.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from remoulade.api.main import app
+from remoulade.state import StateNamesEnum
+
+
+@pytest.fixture
+def api_client(state_middleware):
+    with app.test_client() as client:
+        yield client
+
+
+class TestMessageStateAPI:
+    """ Class Responsible to do the test of the API """
+
+    def test_no_messages(self, stub_broker, state_middleware, api_client):
+        res = api_client.get("/messages/states/")
+        assert res.status_code == 200
+        assert res.data == b"[]\n"
+
+    def test_invalid_url(self, stub_broker, state_middleware, api_client):
+        res = api_client.get("/invalid_url/")
+        assert res.status_code == 404
+
+    def test_pending_message(self, stub_broker, state_middleware, do_work, api_client):
+        do_work.send()
+        res = api_client.get("/messages/states/")
+        for state_enum in StateNamesEnum:
+            if state_enum.name == "Pending":
+                assert state_enum.name.encode() in res.data
+            else:
+                assert state_enum.name.encode() not in res.data
+
+    def test_invalid_state_message(self, stub_broker, do_work, api_client):
+        do_work.send()
+        res = api_client.get("/messages/states/?name=invalid_state")
+        assert b"Bad Request" in res.data
+
+    def test_success_message(self, stub_broker, stub_worker, state_middleware, do_work, api_client):
+        do_work.send()
+        stub_broker.join(do_work.queue_name)
```

Comment:
Do not create a state by enqueing a message it has already been tested elsewhere, you can add it directly with state_middleware.backend like this.

```
                for message_id in range(1500):
            state_middleware.backend.set_state(State(message_id, StateNamesEnum.Success, [], {}), 1000)

        res = api_client.get("/messages/states/")
        assert len(res.json) == 1500
```

I wanted to test we were returning more than 1000 states.

Reply:
with this methods you could add some more tests